lib/build.py

```diff
@@ -976,6 +999,53 @@ def __init__(self, *args):
       self.value = self.tree.terminals_cat("LINE_CHUNK")
 
 
+class Label(Instruction):
+
+   __slots__ = ("key",
+                "value")
+
+   def __init__(self, *args):
+      super().__init__(*args)
+      self.commit_files |= {ch.Path("ch/metadata.json")}
+
+   @property
+   def str_(self):
+      return "%s='%s'" % (self.key, self.value)
+
+   def execute(self):
+      with (self.image.unpack_path // "/ch/metadata.json").open_("wt") \
+           as fp:
+         for (k, v) in self.label_label.items():
+            print("%s=%s" % (k, v), file=fp)
+
+   def prepare(self, *args):
+      self.value = ch.variables_sub(unescape(self.value), self.label_build)
+      self.label_label[self.key] = self.value
+      return super().prepare(*args)
```

Comment:
This seems muddled. I'd expect it to set the image metadata, which I think it does but it's very roundabout?

`self.label_build` should I think be `self.env_build` if that clarifies things.

Reply:
Ok, I think I got some changes figured out and it appears to pass CI. My only question is regarding env_build. I assume it is ok to use this variable even though it is a reference to an env property in the label class? It works, but may be a little indirect. I tried just removing that line of code and this line just helps the grammar interpret the value correctly, so maybe there is another way to do it. If you are happy with the current status though, I'm happy with it too.

Reply:
Yes, you can reference `env_build`. It just means "environment to use during the build". Lots of instructions need that.

Reply:
Great! Thanks